File: tests/algorithms/transpiler/FS/other/alternative_list_arrange.fs
// Generated 2025-08-09 10:14 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _dictAdd<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) (v:'V) =
    d.[k] <- v
    d
let _dictCreate<'K,'V when 'K : equality> (pairs:('K * 'V) list) : System.Collections.Generic.IDictionary<'K,'V> =
    let d = System.Collections.Generic.Dictionary<'K, 'V>()
    for (k, v) in pairs do
        d.[k] <- v
    upcast d
let _dictGet<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) : 'V =
    match d.TryGetValue(k) with
    | true, v -> v
    | _ -> Unchecked.defaultof<'V>
let _idx (arr:'a array) (i:int) : 'a =
    if not (obj.ReferenceEquals(arr, null)) && i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let rec _str v =
    let s = sprintf "%A" v
    let s = if s.EndsWith(".0") then s.Substring(0, s.Length - 2) else s
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
type Item =
    | Int of int
    | Str of string
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec from_int (x: int) =
    let mutable __ret : Item = Unchecked.defaultof<Item>
    let mutable x = x
    try
        __ret <- Int(x)
        raise Return
        __ret
    with
        | Return -> __ret
let rec from_string (s: string) =
    let mutable __ret : Item = Unchecked.defaultof<Item>
    let mutable s = s
    try
        __ret <- Str(s)
        raise Return
        __ret
    with
        | Return -> __ret
let rec item_to_string (it: Item) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable it = it
    try
        __ret <- unbox<string> ((match it with
            | Int(v) -> _str (v)
            | Str(s) -> s))
        raise Return
        __ret
    with
        | Return -> __ret
let rec alternative_list_arrange (first: Item array) (second: Item array) =
    let mutable __ret : Item array = Unchecked.defaultof<Item array>
    let mutable first = first
    let mutable second = second
    try
        let len1: int = Seq.length (first)
        let len2: int = Seq.length (second)
        let abs_len: int = if len1 > len2 then len1 else len2
        let mutable result: Item array = Array.empty<Item>
        let mutable i: int = 0
        while i < abs_len do
            if i < len1 then
                result <- Array.append result [|(_idx first (int i))|]
            if i < len2 then
                result <- Array.append result [|(_idx second (int i))|]
            i <- i + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
let rec list_to_string (xs: Item array) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable xs = xs
    try
        let mutable s: string = "["
        let mutable i: int = 0
        while i < (Seq.length (xs)) do
            s <- s + (item_to_string (_idx xs (int i)))
            if i < ((Seq.length (xs)) - 1) then
                s <- s + ", "
            i <- i + 1
        s <- s + "]"
        __ret <- s
        raise Return
        __ret
    with
        | Return -> __ret
let example1: Item array = alternative_list_arrange (unbox<Item array> [|from_int (1); from_int (2); from_int (3); from_int (4); from_int (5)|]) (unbox<Item array> [|from_string ("A"); from_string ("B"); from_string ("C")|])
printfn "%s" (list_to_string (example1))
let example2: Item array = alternative_list_arrange (unbox<Item array> [|from_string ("A"); from_string ("B"); from_string ("C")|]) (unbox<Item array> [|from_int (1); from_int (2); from_int (3); from_int (4); from_int (5)|])
printfn "%s" (list_to_string (example2))
let example3: Item array = alternative_list_arrange (unbox<Item array> [|from_string ("X"); from_string ("Y"); from_string ("Z")|]) (unbox<Item array> [|from_int (9); from_int (8); from_int (7); from_int (6)|])
printfn "%s" (list_to_string (example3))
let example4: Item array = alternative_list_arrange (unbox<Item array> [|from_int (1); from_int (2); from_int (3); from_int (4); from_int (5)|]) (Array.empty<Item>)
printfn "%s" (list_to_string (example4))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

File: tests/algorithms/x/PHP/other/alternative_list_arrange.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function from_int($x) {
  global $example1, $example2, $example3, $example4;
  return ['__tag' => 'Int', 'value' => $x];
}
function from_string($s) {
  global $example1, $example2, $example3, $example4;
  return ['__tag' => 'Str', 'value' => $s];
}
function item_to_string($it) {
  global $example1, $example2, $example3, $example4;
  return (function($__v) {
  if ($__v['__tag'] === "Int") {
    $v = $__v["value"];
    return _str($v);
  } elseif ($__v['__tag'] === "Str") {
    $s = $__v["value"];
    return $s;
  }
})($it);
}
function alternative_list_arrange($first, $second) {
  global $example1, $example2, $example3, $example4;
  $len1 = count($first);
  $len2 = count($second);
  $abs_len = ($len1 > $len2 ? $len1 : $len2);
  $result = [];
  $i = 0;
  while ($i < $abs_len) {
  if ($i < $len1) {
  $result = _append($result, $first[$i]);
}
  if ($i < $len2) {
  $result = _append($result, $second[$i]);
}
  $i = $i + 1;
};
  return $result;
}
function list_to_string($xs) {
  global $example1, $example2, $example3, $example4;
  $s = '[';
  $i = 0;
  while ($i < count($xs)) {
  $s = $s . item_to_string($xs[$i]);
  if ($i < count($xs) - 1) {
  $s = $s . ', ';
}
  $i = $i + 1;
};
  $s = $s . ']';
  return $s;
}
$example1 = alternative_list_arrange([from_int(1), from_int(2), from_int(3), from_int(4), from_int(5)], [from_string('A'), from_string('B'), from_string('C')]);
echo rtrim(list_to_string($example1)), PHP_EOL;
$example2 = alternative_list_arrange([from_string('A'), from_string('B'), from_string('C')], [from_int(1), from_int(2), from_int(3), from_int(4), from_int(5)]);
echo rtrim(list_to_string($example2)), PHP_EOL;
$example3 = alternative_list_arrange([from_string('X'), from_string('Y'), from_string('Z')], [from_int(9), from_int(8), from_int(7), from_int(6)]);
echo rtrim(list_to_string($example3)), PHP_EOL;
$example4 = alternative_list_arrange([from_int(1), from_int(2), from_int(3), from_int(4), from_int(5)], []);
echo rtrim(list_to_string($example4)), PHP_EOL;
